src/main/java/org/folio/circulation/domain/notice/schedule/DueDateNotRealTimeScheduledNoticeHandler.java

```diff
@@ -79,13 +88,26 @@ public DueDateNotRealTimeScheduledNoticeHandler(
   }
 
   private CompletableFuture<Result<Pair<ScheduledNotice, LoanAndRelatedRecords>>> getContext(ScheduledNotice notice) {
-    return loanRepository.getById(notice.getLoanId())
+
+    String templateId = notice.getConfiguration().getTemplateId();
+
+    return templateRepository.findById(templateId)
+      .thenApply(r -> r.next(response -> failIfTemplateNotFound(response, templateId)))
```

Comment:
How does failing to find the notice trigger the notice to be deleted? It isn't obvious to me from this code. Does any failure in the sequence cause the notice to be deleted?

Reply:
the notice will be removed two lines below: 
`   .thenCompose(r -> dueDateScheduledNoticeHandler.deleteNoticeIfLoanIsMissingOrIncomplete(r, notice))`

Reply:
Ok, I find this difficult to follow, however I shall resolve this comment.